public/app/features/alerting/unified/useRouteGroupsMatcher.ts

```diff
@@ -0,0 +1,89 @@
+import * as comlink from 'comlink';
+import { useCallback, useEffect } from 'react';
+import { useEnabled } from 'react-enable';
+
+import { logError } from '@grafana/runtime';
+
+import { AlertmanagerGroup, RouteWithID } from '../../../plugins/datasource/alertmanager/types';
+
+import { logInfo } from './Analytics';
+import { createWorker } from './createRouteGroupsMatcherWorker';
+import { AlertingFeature } from './features';
+import type { RouteGroupsMatcher } from './routeGroupsMatcher.worker';
+
+let routeMatcher: comlink.Remote<RouteGroupsMatcher> | undefined;
+
+// Load worker loads the worker if it's not loaded yet
+// and returns a function to dispose of the worker
+// We do it to enable feature toggling. If the feature is disabled we don't wont to load the worker code at all
+// An alternative way would be to move all this code to the hook below, but it will create and terminate the worker much more often
+function loadWorker() {
+  let worker: Worker | undefined;
+
+  if (routeMatcher === undefined) {
+    try {
+      worker = createWorker();
+      routeMatcher = comlink.wrap<RouteGroupsMatcher>(worker);
+    } catch (e: unknown) {
+      if (e instanceof Error) {
+        logError(e);
+      }
+      console.error('CANNOT LOAD WORKER', e);
```

Comment:
Do we need this if we're already calling `logError`?

Reply:
No, I'll remove